test/TestProjects/TypeSchemaMapping/readme.md

```diff
@@ -0,0 +1,16 @@
+# TypeSchemaMapping
+
+## AutoRest Configuration
+
+> see https://aka.ms/autorest
+
+``` yaml
+require: $(this-folder)/../../../readme.md
+input-file: $(this-folder)/TypeSchemaMapping.json
+namespace: TypeSchemaMapping
+output-folder: $(this-folder)/SomeFolder/Generated
+skip-csproj: true
+project-folder: $(this-folder)
+# relative path to output-folder also works
+# project-folder: ../../
```

Comment:
@AlexanderSher I updated the PR to support both absolute path and relative path for `project-folder`. To make things easier, in the constructor of Configuration, I will turn the absolute path to relative path, which requires no changes to existing projectDirectory handling (except the option renaming) and also meets the need that `Configuration.json` should only contain relative path (absolute path are not portable to different machines/environments).